app/lib/test_applications.rb

```diff
@@ -1,12 +1,22 @@
 module TestApplications
-  def self.create_application(states)
+  def self.create_application(states:)
     first_name = Faker::Name.unique.first_name
     last_name = Faker::Name.unique.last_name
     candidate = FactoryBot.create(
       :candidate,
       email_address: "#{first_name.downcase}.#{last_name.downcase}@example.com",
     )
 
+    course_pool = Course.joins(:course_options).where(open_on_apply: true).all
```

Comment:
I think you could golf this down to:

```suggestion
    courses_to_apply_to = Course.joins(:course_options).where(open_on_apply: true).order('RANDOM()').limit(states.count)
```

Reply:
Yes — more efficient too